crates/augurs-forecaster/src/power_transforms.rs

```diff
@@ -0,0 +1,70 @@
+use crate::transforms::box_cox;
+use argmin::core::*;
+use argmin::solver::brent::BrentOpt;
+
+fn box_cox_log_likelihood(data: &[f64], lambda: f64) -> f64 {
+    let n = data.len() as f64;
+    let transformed_data: Vec<f64> = data.iter().map(|&x| box_cox(x, lambda)).collect();
+    let mean_transformed: f64 = transformed_data.iter().copied().sum::<f64>() / n;
+    let variance: f64 = transformed_data
+        .iter()
+        .map(|&x| (x - mean_transformed).powi(2))
+        .sum::<f64>()
+        / n;
+
+    // Avoid log(0) by ensuring variance is positive
+    let log_likelihood =
+        -0.5 * n * variance.ln() + (lambda - 1.0) * data.iter().map(|&x| x.ln()).sum::<f64>();
+    log_likelihood
+}
+
+#[derive(Clone)]
+struct BoxCoxProblem {
+    data: Vec<f64>,
+}
+
+impl CostFunction for BoxCoxProblem {
+    type Param = f64;
+    type Output = f64;
+
+    // The goal is to minimize the negative log-likelihood
+    fn cost(&self, lambda: &Self::Param) -> Result<Self::Output, Error> {
+        Ok(-box_cox_log_likelihood(&self.data, *lambda))
+    }
+}
+
+/// Optimize the lambda parameter for the Box-Cox transformation
+pub fn optimize_lambda(data: &[f64]) -> f64 {
+    let cost = BoxCoxProblem {
+        data: data.to_vec(),
+    };
+    let init_param = 0.5;
+    let solver = BrentOpt::new(-2.0, 2.0);
+
+    let res = Executor::new(cost, solver)
+        .configure(|state| state.param(init_param).max_iters(100))
+        .run()
+        .unwrap();
+    res.state.best_param.unwrap()
+}
```

Comment:
This looks like it can fail in a couple of ways; we should return a `Result<f64, argmin::Error>` instead to avoid the `unwrap()` calls, and let the caller handle the error.